### PR TITLE
build(sharejs-json1): convert to fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
semi-auto

### CODE REVIEW

File: fluidBuild.config.cjs [hunk 1]
```diff
@@ -84,7 +84,9 @@ module.exports = {
 		"depcruise": [],
 		"check:release-tags": ["tsc"],
 		"check:are-the-types-wrong": ["build"],
-		"eslint": [...tscDependsOn, "commonjs"],
+		// ADO #7297: Review why the direct dependency on 'build:esm:test' is necessary.
+		//            Should 'compile' be enough?  compile -> build:test -> build:test:esm
+		"eslint": ["compile", "build:test:esm"],
```
Comment (DLehenbauer) [inline fluidBuild.config.cjs:89]: Workaround for AB#7297.

**Review by sonalideshpandemsft — APPROVED** [review]

**Review by jason-ha — APPROVED** [review]

File: experimental/dds/ot/sharejs/json1/package.json [hunk 2]
```diff
@@ -36,9 +72,11 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../../../.prettierignore",
 		"test": "npm run test:mocha",
 		"test:coverage": "c8 npm test",
-		"test:mocha": "mocha --recursive dist/test -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha": "npm run test:mocha:cjs && npm run test:mocha:esm",
+		"test:mocha:cjs": "mocha  --recursive \"dist/test/*.spec.*js\" --exit --project src/test/tsconfig.cjs.json -r node_modules/@fluid-internal/mocha-test-setup",
```
Comment (jason-ha) [inline experimental/dds/ot/sharejs/json1/package.json:76]: Do you have any idea about the `--project` mocha option. It is not documented at https://mochajs.org/#command-line-usage and our projects are special with the dual build. If this isn't doing anything, we should remove it. I also wonder about `-r` but don't have connection to config to worry about.

Comment (jason-ha) [reply experimental/dds/ot/sharejs/json1/package.json:76]: @DLehenbauer , @sonalideshpandemsft ^^ don't want this lost in the conversions. We should understand mocha --project. Is it real?

Comment (jason-ha) [reply experimental/dds/ot/sharejs/json1/package.json:76]: Oh, looks like #19774 aims to remove these. Nice :)

**Pull request merged**